contributivity_measures.py

```diff
@@ -168,3 +166,704 @@ def not_twice_characteristic(permut):
+            "std_sv": np.array([0]),
+            "prop": np.array([1]),
+            "computed_val": char_value_dict,
+        }
+    else:
+
+        # definition of the original density
+        def prob(subset):
+            lS = len(subset)
+            return factorial(n - 1 - lS) * factorial(lS) / factorial(n)
+
+        # definition of the approximation of the increment
+        # ## compute the last and the first increments in performance (they are needed to compute the approximated increments)
+        characteristic_no_nodes = 0
+        last_increments = []
+        first_increments = []
+        for k in range(n):
+            last_increments.append(
+                characteristic_all_node
+                - not_twice_characteristic(np.delete(np.arange(n), k))
+            )
+            first_increments.append(
+                not_twice_characteristic(np.array([k])) - characteristic_no_nodes
+            )
+
+        # ## definition of the number of data in all datasets
+        size_of_I = 0
+        for node in preprocessed_node_list:
+            size_of_I += len(node.y_train)
+
+        def approx_increment(subset, k):
+            assert k not in subset, "" + str(k) + "is not in " + str(subset) + ""
+            small_node_list = np.array([preprocessed_node_list[i] for i in subset])
+            # compute the size of subset : ||subset||
+            size_of_S = 0
+            for node in small_node_list:
+                size_of_S += len(node.y_train)
+            beta = size_of_S / size_of_I
+            return (1 - beta) * first_increments[k] + beta * last_increments[k]
+
+        # compute  the importance density
+        # ## compute the renormalization constant of the importance density for all datatsets
+
+        renorms = []
+        for k in range(n):
+            list_k = np.delete(np.arange(n), k)
+            renorm = 0
+            for length_combination in range(len(list_k) + 1):
+                for subset in combinations(
+                    list_k, length_combination
+                ):  # could be avoided as   prob(np.array(subset))*np.abs(approx_increment(np.array(subset),j)) is constant in the combination
+                    renorm += prob(np.array(subset)) * np.abs(
+                        approx_increment(np.array(subset), k)
+                    )
+            renorms.append(renorm)
+
+        # ## defines the importance density
+        def g(subset, k):
+            return (
+                prob(np.array(subset))
+                * np.abs(approx_increment(np.array(subset), k))
+                / renorms[k]
+            )
+
+        # sampling
+        t = 0
+        q = -norm.ppf((1 - alpha) / 2, loc=0, scale=1)
+        v_max = 0
+        while (
+            t < 100 or t < 4 * q ** 2 * v_max / (sv_accuracy) ** 2
+        ):  # Check if the length of the confidence interval  is below the value of sv_accuracy*characteristic_all_node
+            t += 1
+            print(t)
```

Comment:
The prinds floods the output. We can remove the print or print every 1000th iteration for instance ?